src/main/scala/com/lightbend/conductr/sbt/BundlePlugin.scala

```diff
@@ -395,93 +424,74 @@ object BundlePlugin extends AutoPlugin {
   }
 
   private def getConfig(config: Configuration, forAllSettings: Boolean): Def.Initialize[Task[String]] = Def.task {
-    val checkComponents = (checksConfigName in config).value match {
-      case (value, configName) if (forAllSettings || configName.isDefined) && value.nonEmpty =>
-        val checkInitialDelayValue = (checkInitialDelay in config).value
-        val checkInitialDelayInSeconds =
-          if (checkInitialDelayValue.toMillis % 1000 > 0)
-            checkInitialDelayValue.toSeconds + 1 // always round up
-          else
-            checkInitialDelayValue.toSeconds
-        Seq(
-          value.map(uri => s""""$uri"""").mkString(
-            s"""components = {
-                |  ${(normalizedName in config).value}-status = {
-                |    description      = "Status check for the bundle component"
-                |    file-system-type = "universal"
-                |    start-command    = ["check", "--initial-delay", "$checkInitialDelayInSeconds", """.stripMargin,
-            ", ",
-            s"""]
-                |    endpoints        = {}
-                |  }
-                |}""".stripMargin
+    def valueSuppliedForConfig[T](valueAndConfigName: (T, Option[String])): Boolean =
+      forAllSettings || valueAndConfigName._2.isDefined
+
+    if (valueSuppliedForConfig((normalizedNameConfigName in config).value) ||
+      valueSuppliedForConfig((compatibilityVersionConfigName in config).value) ||
+      valueSuppliedForConfig((systemConfigName in config).value) ||
+      valueSuppliedForConfig((systemVersionConfigName in config).value) ||
+      valueSuppliedForConfig((nrOfCpusConfigName in config).value) ||
+      valueSuppliedForConfig((memoryConfigName in config).value) ||
+      valueSuppliedForConfig((diskSpaceConfigName in config).value) ||
+      valueSuppliedForConfig((rolesConfigName in config).value) ||
+      valueSuppliedForConfig((projectInfoConfigName in config).value) ||
+      valueSuppliedForConfig((bundleTypeConfigName in config).value) ||
+      valueSuppliedForConfig((startCommandConfigName in config).value) ||
+      valueSuppliedForConfig((endpointsConfigName in config).value)) {
+
+      val checkComponents = (checksConfigName in config).value match {
+        case (value, configName) if (forAllSettings || configName.isDefined) && value.nonEmpty =>
+          val checkInitialDelayValue = (checkInitialDelay in config).value
+          val checkInitialDelayInSeconds = Math.max(1, checkInitialDelayValue.toSeconds)
```

Comment:
Looking at it again, I just want to ensure that it isn’t zero.

> On 25 May 2016, at 3:52 PM, Mirco Dotta <notifications@github.com> wrote:
> 
> In src/main/scala/com/lightbend/conductr/sbt/BundlePlugin.scala <https://github.com/typesafehub/sbt-conductr/pull/156#discussion_r64518516>:
> 
> > +      valueSuppliedForConfig((compatibilityVersionConfigName in config).value) ||
> > +      valueSuppliedForConfig((systemConfigName in config).value) ||
> > +      valueSuppliedForConfig((systemVersionConfigName in config).value) ||
> > +      valueSuppliedForConfig((nrOfCpusConfigName in config).value) ||
> > +      valueSuppliedForConfig((memoryConfigName in config).value) ||
> > +      valueSuppliedForConfig((diskSpaceConfigName in config).value) ||
> > +      valueSuppliedForConfig((rolesConfigName in config).value) ||
> > +      valueSuppliedForConfig((projectInfoConfigName in config).value) ||
> > +      valueSuppliedForConfig((bundleTypeConfigName in config).value) ||
> > +      valueSuppliedForConfig((startCommandConfigName in config).value) ||
> > +      valueSuppliedForConfig((endpointsConfigName in config).value)) {
> > +
> > +      val checkComponents = (checksConfigName in config).value match {
> > +        case (value, configName) if (forAllSettings || configName.isDefined) && value.nonEmpty =>
> > +          val checkInitialDelayValue = (checkInitialDelay in config).value
> > +          val checkInitialDelayInSeconds = Math.max(1, checkInitialDelayValue.toSeconds)
> Is the value of checkInitialDelayValue.toSeconds round up? Just asking as otherwise there is a small difference with the former implementation
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/typesafehub/sbt-conductr/pull/156/files/dc595fa371c928fd8de73be5ab58cdaa280bbbbd#r64518516>